hydpy/models/rconc/rconc_model.py

```diff
@@ -0,0 +1,275 @@
+        >>> control.uh.shape = 3
+        >>> control.uh = 0.3, 0.5, 0.2
+        >>> logs.quh.shape = 3
+        >>> logs.quh = 1.0, 3.0, 0.0
+
+        Without new input, the actual output is simply the first value stored in the
+        logging sequence, and the values of the logging sequence shift to the left:
+
+        >>> fluxes.inflow = 0.0
+        >>> model.determine_outflow_v1()
+        >>> fluxes.outflow
+        outflow(1.0)
+        >>> logs.quh
+        quh(3.0, 0.0, 0.0)
+
+        With a new input of 4 mm, the actual output consists of the first
+        value stored in the logging sequence and the input value
+        multiplied with the first unit hydrograph ordinate.  The updated
+        logging sequence values result from the multiplication of the
+        input values and the remaining ordinates:
+
+        >>> fluxes.inflow = 4.0
+        >>> model.determine_outflow_v1()
+        >>> fluxes.outflow
+        outflow(4.2)
+        >>> logs.quh
+        quh(2.0, 0.8, 0.0)
+
+        The following example demonstrates the updating of a non-empty logging sequence:
+
+        >>> fluxes.inflow = 4.0
+        >>> model.determine_outflow_v1()
+        >>> fluxes.outflow
+        outflow(3.2)
+        >>> logs.quh
+        quh(2.8, 0.8, 0.0)
+
+        A unit hydrograph consisting of one ordinate routes the received input directly:
+
+        >>> control.uh.shape = 1
+        >>> control.uh = 1.0
+        >>> fluxes.inflow = 0.0
+        >>> logs.quh.shape = 1
+        >>> logs.quh = 0.0
+        >>> model.determine_outflow_v1()
+        >>> fluxes.outflow
+        outflow(0.0)
+        >>> logs.quh
+        quh(0.0)
+        >>> fluxes.inflow = 4.0
+        >>> model.determine_outflow_v1()
+        >>> fluxes.outflow
+        outflow(4.0)
+        >>> logs.quh
+        quh(0.0)
+    """
+
+    CONTROLPARAMETERS = (rconc_control.UH,)
+    DERIVEDPARAMETERS = ()
+    REQUIREDSEQUENCES = (rconc_fluxes.Inflow,)
+    UPDATEDSEQUENCES = (rconc_logs.QUH,)
+    RESULTSEQUENCES = (rconc_fluxes.Outflow,)
+
+    @staticmethod
+    def __call__(model: modeltools.Model) -> None:
+        con = model.parameters.control.fastaccess
+        flu = model.sequences.fluxes.fastaccess
+        log = model.sequences.logs.fastaccess
+        flu.outflow = con.uh[0] * flu.inflow + log.quh[0]
+        for jdx in range(1, len(con.uh)):
+            log.quh[jdx - 1] = con.uh[jdx] * flu.inflow + log.quh[jdx]
+
+
+class Determine_Outflow_V2(modeltools.Method):
+    r"""Calculate the linear storage cascade output (state-space approach).
+
+    Basic equations:
+        :math:`Outflow = KSC \cdot SC`
+
+        :math:`\frac{dSC}{dt} = Inflow - Outflow`
+
+    Note that the given base equations only hold for one single linear storage, while
+    |Determine_Outflow_V2| supports a cascade of linear storages.  Also, the equations
+    do not reflect the possibility to increase numerical accuracy via decreasing the
+    internal simulation step size.
```

Comment:
"to increase numerical accuracy via" -> " of increasing numerical accuracy by"

Reply:
ok